src/Core/NotificationHub/NotificationHubPushNotificationService.cs

```diff
@@ -192,26 +205,36 @@ public async Task PushNotificationAsync(Notification notification)
             ClientType = notification.ClientType,
             UserId = notification.UserId,
             OrganizationId = notification.OrganizationId,
+            InstallationId = installationId,
             Title = notification.Title,
             Body = notification.Body,
             CreationDate = notification.CreationDate,
             RevisionDate = notification.RevisionDate
         };
 
-        if (notification.UserId.HasValue)
+        if (notification.Global && installationId.HasValue)
```

Comment:
Chose to log warning in constructor, since changing installation it to be required and throwing exceptions is large change. I will add it as an option to https://bitwarden.atlassian.net/browse/PM-15513
Fixed in 186a09bb

Reply:
Not exactly what I was asking about. 

```
    if (notification.Global && installationId.HasValue)
        {
            await SendPayloadToInstallationAsync(installationId.Value, PushType.Notification, message, true,
                notification.ClientType);
        }
        else if (notification.UserId.HasValue)
        {
            await SendPayloadToUserAsync(notification.UserId.Value, PushType.Notification, message, true,
                notification.ClientType);
        }
        else if (notification.OrganizationId.HasValue)
        {
            await SendPayloadToOrganizationAsync(notification.OrganizationId.Value, PushType.Notification, message,
                true, notification.ClientType);
        }
        else 
        {
           /// Currently siliently does nothing should we add warning logging here?
        }
    }
```

This would specifically be a miss configured notification which shouldn't happen, but my preference is to handle the unexpected case explicitly with either an exception or a log instead of ignoring it.